aws/logs_monitoring/parsing.py

```diff
@@ -528,6 +528,116 @@ def cwevent_handler(event, metadata):
     yield data
 
 
+def parse_aws_waf_logs(event):
+    """Parse out complex arrays of objects in AWS WAF logs
+
+    Attributes to convert:
+        httpRequest.headers
+        nonTerminatingMatchingRules
+        rateBasedRuleList
+        ruleGroupList
+
+    This prevents having an unparsable array of objects in the final log.
+    """
+    if event.get(DD_SOURCE) != "waf":
+        return event
+    event_copy = copy.deepcopy(event)
+
+    message = event_copy.get("message", {})
+    if isinstance(message, str):
+        try:
+            message = json.loads(message)
+        except json.JSONDecodeError:
+            logger.debug("Failed to decode waf message")
+            return
+
+    headers = message.get("httpRequest", {}).get("headers")
+    if headers:
+        message["httpRequest"]["headers"] = {}
+        for index, item in enumerate(headers):
+            current_header = headers[index]
+            name = current_header["name"]
+            value = current_header["value"]
+            message["httpRequest"]["headers"][name] = value
```

Comment:
NIT: we don't necessarily need the index here (same applies to other places in the code) so we can write this code as follows (also added a couple of checks):

```suggestion
    headers = message.get("httpRequest", {}).get("headers")
    if headers and isinstance(headers, list):
        header_obj = {}
        for header in headers:
            if ('name' in header and 'value' in header):
                header_obj.update({
                    header["name"]: header["value"]
                })
        message["httpRequest"]["headers"] = header_obj
```